bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/query/QueryableUpdates.java

```diff
@@ -43,16 +43,17 @@ public IQueryResult<IInstallableUnit> query(IQuery<IInstallableUnit> query, IPro
 		if (monitor == null)
 			monitor = new NullProgressMonitor();
 		int totalWork = 2000;
-		monitor.beginTask(ProvUIMessages.QueryableUpdates_UpdateListProgress, totalWork);
+		SubMonitor subMonitor = SubMonitor.convert(monitor, ProvUIMessages.QueryableUpdates_UpdateListProgress,
+				totalWork);
 		IPlanner planner = ui.getSession().getProvisioningAgent().getService(IPlanner.class);
 		try {
 			Set<IInstallableUnit> allUpdates = new HashSet<>();
 			for (IInstallableUnit unit : iusToUpdate) {
-				if (monitor.isCanceled())
+				if (subMonitor.isCanceled())
```

Comment:
Same here

Reply:
This whole `if` check can go, given that the `OperationCanceledException` is explicitly caught. In both cases, `Collector.emptyCollector()` is returned.